dolphinscheduler-task-plugin/dolphinscheduler-task-java/src/main/java/org/apache/dolphinscheduler/plugin/task/java/JavaTask.java

```diff
@@ -0,0 +1,422 @@
+    @Override
+    public void init() {
+        logger.info("java task params {}", taskRequest.getTaskParams());
+        javaParameters = JSONUtils.parseObject(taskRequest.getTaskParams(), JavaParameters.class);
+        if (javaParameters == null || !javaParameters.checkParameters()) {
+            throw new TaskException("java task params is not valid");
+        }
+        if (javaParameters.getRunType().equals(JavaConstants.RUN_TYPE_JAR)) {
+            setMainJarName();
+        }
+    }
+
+    /**
+     * @description: Gets the Java source file that was initially processed
+     * @param: []
+     * @return: java.lang.String
+     **/
+    @Override
+    public String getPreScript() {
+        String rawJavaScript = javaParameters.getRawScript().replaceAll("\\r\\n", "\n");
+        try {
+            rawJavaScript = convertJavaSourceCodePlaceholders(rawJavaScript);
+        } catch (StringIndexOutOfBoundsException e) {
+            logger.error("setShareVar field format error, raw java script : {}", rawJavaScript);
+        }
+        return rawJavaScript;
+    }
+
+    /**
+     * @description: Execute Java tasks
+     * @param: []
+     * @return: void
+     **/
+    @Override
+    public void handle() throws Exception {
+        try {
+            // Step 1: judge if is java or jar run type.
+            // Step 2 case1: The jar run type builds the command directly, adding resource to the java -jar class when building the command
+            // Step 2 case2: The java run type, first replace the custom parameters, then compile the code, and then build the command will add resource
+            // Step 3: To run the command
+            String command = null;
+            switch (javaParameters.getRunType()) {
+                case JavaConstants.RUN_TYPE_JAVA:
+                    command = buildJavaCommand();
+                    break;
+                case JavaConstants.RUN_TYPE_JAR:
+                    command = buildJarCommand();
+                    break;
+                default:
+                    throw new RunTypeNotFoundException("run type is required, but it is null now.");
+            }
+            Preconditions.checkNotNull(command, "command not be null.");
+            TaskResponse taskResponse = shellCommandExecutor.run(command);
+            logger.info("java task run result : {}", taskResponse);
+            setExitStatusCode(taskResponse.getExitStatusCode());
+            setAppIds(taskResponse.getAppIds());
+            setProcessId(taskResponse.getProcessId());
+            setVarPool(shellCommandExecutor.getVarPool());
+        } catch (InterruptedException e) {
+            logger.error("java task interrupted ", e);
+            setExitStatusCode(TaskConstants.EXIT_CODE_FAILURE);
+            Thread.currentThread().interrupt();
+        } catch (RunTypeNotFoundException e) {
+            logger.error(e.getMessage());
+            setExitStatusCode(TaskConstants.EXIT_CODE_FAILURE);
+            throw e;
```

Comment:
```suggestion
        
```
This catch is not needed, if you want to print the exception, it's better to print the thread stack rather than e.getMessage.